hash.go

```diff
@@ -358,6 +359,24 @@ func (h *md5Marshal) UnmarshalBinary(b []byte) error {
 	return nil
 }
 
+func (h *md5Marshal) AppendBinary(buf []byte) ([]byte, error) {
+	binaryData, err := h.MarshalBinary()
+	if err != nil {
+		return nil, err
+	}
+	return append(buf, binaryData...), nil
+}
+
+func (h *md5Marshal) AppendText(buf []byte) ([]byte, error) {
```

Comment:
The standard library hashers don't implement `encoding.AppendText`, so there is no need for us to implement that on our side. The less code we maintain, the better.